Initial kayobe configuration for SIB biomed

jovial opened the pull request.

markgoddard committed:
Use the overlay storage driver for docker

This avoids the need for a block device (as required by devicemapper).

markgoddard committed:
Use a remote virtual environment for kayobe overcloud operations

markgoddard committed:
Set the bootstrap user for controllers, compute and storage groups

The centos user is added by cloud-init on boot.

markgoddard committed:
Don't use LVM on compute or storage nodes

There is only a single disk, which is used for a single root partition.
We're using overlayfs, so an LV is not required for docker storage.

jovial committed:
Add etcd synchronization for neutron genericswitch driver

This allows concurrent connections to the switch to be limited.

markgoddard committed:
Configure timeouts and concurrency for Neutron generic switch

The management switches take a long time to configure (~20 seconds per
operation).  They can also suffer from configuration failures when multiple
sessions concurrently attempt to modify the configuration, resulting in the
following message:

Configuration Database locked by another application - try later

This failure will not be detected by the generic switch driver.

We aim to provide a trade off between reliability and timely configuration by
allowing up to 5 concurrent configuration sessions. With a large number of
concurrent requests, this may still result in long delays - roughly 15
operations can be processed per minute per switch.

To allow for this slow progress, we increase various timeouts associated
with port operations to 5 minutes, allowing for bulk processing of up to 75
operations. Note that deploying a baremetal node actually requires three
port operations - plug to provisioning network, then unplug and plug to the
tenant network.

jovial committed:
Add reference to project in readme

jovial committed:
configure local ntp server

jovial committed:
Use local DNS server

jovial committed:
Rename example hosts file

jovial committed:
Add seed node to the host inventory

jovial committed:
Enable periodic yum security updates via cron

jovial committed:
Add a public key for user Will Szumski

The user will authorised for SSH as the stack and kolla users.

jovial committed:
Configure IPA image build

IPA images will be built locally. The image will include support for collecting
additional information during hardware inspection via the 'extra-hardware'
collector. It also includes some custom hardware managers with support
for verifying BIOS and NIC firmware versions.

jovial committed:
Add configuration of IPMI username and password for BMCs

jovial committed:
Add configuration for ironic inspector

We enable some processing hooks:

* system_name_physnet to set physical_network in ironic port
* extra_hardware to process additional inventory data collected by the
  extra-hardware collector

We also add all discovered ports to ironic.

jovial committed:
Use StackHPC forks for kolla and kolla ansible

Our forks contain various bug fixes that we have backported to
the pike release.

jovial committed:
Add initial network definitions

Currently includes:

* oc_provision - overcloud provisioning network
* wl_provision - workload provisioning network
* internal - internal API network

jovial committed:
Add network interface config for seed

The seed is attached to the overcloud provisioning network.

jovial committed:
Use StackHPC fork of bifrost

This tag is based on upstream stable/pike, but with
some backported fixes.

jovial committed:
Don't configure resolv.conf on the seed

We'll use the configuration obtained via DHCP.

jovial committed:
Use a private Docker registry

A registry has been created on the seed.

jovial committed:
Add control switch to inventory

Adds configuration file skeleton.

jovial committed:
Do not create any LVM volumes for controller

these are not required as the devicemapper docker storage driver
is not in use (we're using overlay)

jovial committed:
Add storage and storage management networks

jovial committed:
add overcloud and workload management networks

jovial committed:
Add initial network interface configuration for controllers

jovial committed:
Configure neutron dhcp agent to use local dns servers

jovial committed:
tweaks to nova baremetal compute service

jovial committed:
Initial configuration of neutron

- We need to update network generic switch to support the
  Dell powerconnect 6224 switch that we are using.

jovial committed:
Configure ironic inspector

* Remove the scheduler hook from ironic inspector processing pipeline.
  This applies scheduling properties to nodes during inspection. We
  don't want this as it could mean that VMs get scheduled to bare metal nodes.

jovial committed:
fix indentation

jovial committed:
Add baremetal compute hosts to inventory

jovial committed:
Add initial switch configuration

jovial committed:
Add powerconnect 6224 port mappings

Uses the new dell-powerconnect-switch role in kayobe. We will
need to port this to pike.

markgoddard commented on etc/kayobe/inspector.yml:
```diff
@@ -19,16 +19,27 @@
 # List of of additional inspector processing plugins.
 #inspector_processing_hooks_extra:
+inspector_processing_hooks_extra:
+  - system_name_physnet
```
If you add system_name_llc here it will add the switch hostname to ironic ports' local_link_connection/switch_info, then you can refer to switches by hostname in NGS rather than MAC.

jovial replied:
Added

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -94,7 +95,13 @@
+kolla_build_customizations:
+  neutron_server_footer: |
+    # Need a version of networking generic switch with dell powerconnect support
+    RUN pip install git+https://github.com/stackhpc/networking-generic-switch@stackhpc/1.0.0.1
```
This release is based on stackhpc/queens. It'll probably work, but if not you might need to backport the change to stackhpc/pike.

jovial committed:
Adjust switch config for PowerConnect 6224 compatability

The Dell PowerConnect 6224 differs from the Dell PowerConnect 5448
we have used before. It needs the following changes:

* You need to enter privilege exec mode prior to switch configuration
* port identifiers take a different form

jovial committed:
Powerconnect 6224 use transmit-tlv instead of optional-tlv

markgoddard commented on etc/kayobe/inventory/group_vars/controllers/nova-compute-ironic.conf:
```diff
@@ -0,0 +1,8 @@
+
+[compute]
+# Don't disable the compute service due to failed builds.
+consecutive_build_service_disable_threshold = 0
```
This goes in etc/kayobe/kolla/config/nova/nova-compute-ironic.conf

markgoddard commented on etc/kayobe/inventory/group_vars/controllers/network-interfaces:
```diff
@@ -2,35 +2,28 @@
+wl_provision_interface: "{{ oc_provision_interface }}.{{ wl_provision_vlan }}"
+
+# Internal network interface.
+internal_interface: "{{ oc_provision_interface }}.{{ internal_vlan }}"
```
Public interface required?

jovial replied:
We do not have a public network defined at this moment in time

markgoddard commented on etc/kayobe/inventory/group_vars/controllers/network-interfaces:
```diff
@@ -2,35 +2,28 @@
+wl_oob_interface: "{{ oc_provision_interface }}.{{ wl_oob_vlan }}"
+
+# Workload provisioning network interface.
+wl_provision_interface: "{{ oc_provision_interface }}.{{ wl_provision_vlan }}"
```
Are there bare metal compute nodes?

jovial replied:
Yep, we have 8 bare metal compute nodes.

markgoddard commented on etc/kayobe/inventory/group_vars/seed/dns:
```diff
@@ -0,0 +1,3 @@
+---
+# Don't manage resolv.conf - use the DHCP response from neutron on VSS.
```
We probably don't want this if we're managing the seed node's hypervisor (are we?)

jovial replied:
Not 100% sure this. I thought we were were using a dedicated server for the seed.

markgoddard commented on etc/kayobe/inventory/group_vars/seed/network-interfaces:
```diff
@@ -2,35 +2,10 @@
+# Overcloud provisioning network interface.
+oc_provision_interface: eth1
+# Use DHCP on VSS for this interface.
+oc_provision_bootproto: dhcp
```
We probably don't want this if we're managing the seed node's hypervisor (are we?)

jovial replied:
I believe we have a dedicated server for the seed node

markgoddard commented on etc/kayobe/kolla.yml:
```diff
@@ -59,9 +59,10 @@
 
 # URL of docker registry to use for Kolla images.
 #kolla_docker_registry:
+kolla_docker_registry: "{{ docker_registry }}"
```
We don't currently have automation for deploying a registry on the seed, but it's easy enough to do. I think the commands are in the eod deploy notes.

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
 #public_net_name:
```
Are you using a single network for public and internal APIs?

jovial replied:
The public network has been left out of the design specification at this moment in time.

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -7,21 +7,26 @@
 # Name of the network used by the overcloud hosts to manage the bare metal
 # compute hosts via their out-of-band management controllers.
 #oob_wl_net_name:
+oob_wl_net_name: wl_oob
```
Unless specifically requested, it might be easier to use a single VLAN for both compute and control OOB.

jovial replied:
In the questionnaire document - which has morphed into the design specification - these are separated.

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+# Pool for use by bifrost's dnsmasq service during hardware inspection and
+# provisioning.
+oc_provision_inspection_allocation_pool_start: 10.12.3.128
+oc_provision_inspection_allocation_pool_end: 10.12.3.255
```
Finish at .254 since that's the end of the subnet.

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+wl_oob_cidr: 10.12.7.0/16
+# Pool for use by kayobe.
+wl_oob_allocation_pool_start: 10.12.7.0
+wl_oob_allocation_pool_end: 10.12.7.255
```
.254

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+internal_cidr: 10.12.4.0/24
+# Pool for use by kayobe.
+internal_allocation_pool_start: 10.12.4.2
+internal_allocation_pool_end: 10.12.4.255
```
.254

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+storage_cidr: 10.12.5.0/24
+# Pool for use by kayobe.
+storage_allocation_pool_start: 10.12.5.0
+storage_allocation_pool_end: 10.12.5.255
```
.254

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+storagemgmt_cidr: 10.12.6.0/24
+# Pool for use by kayobe.
+storagemgmt_allocation_pool_start: 10.12.6.0
+storagemgmt_allocation_pool_end: 10.12.6.255
```
.254

markgoddard commented on etc/kayobe/networks.yml:
```diff
@@ -34,107 +39,89 @@
+oc_provision_inspection_allocation_pool_end: 10.12.3.255
+#oc_provision_net_gateway:
+oc_provision_vlan: 73
+oc_provision_physical_network: "{{ sib_mgmt_physical_network }}"
```
Did you say this was on a different network?

markgoddard commented on etc/kayobe/neutron.yml:
```diff
@@ -4,7 +4,12 @@
-#kolla_neutron_ml2_mechanism_drivers:
+kolla_neutron_ml2_mechanism_drivers:
+  kolla_neutron_ml2_mechanism_drivers:
+  - mlnx_sdn_assist
```
Is there an IB network?

markgoddard commented on etc/kayobe/storage.yml:
```diff
@@ -54,6 +55,7 @@
 # List of storage volume groups. See mrlesmithjr.manage-lvm role for
 # format.
 #storage_lvm_groups:
+storage_lvm_groups: []
```
We should look at making this the default in kayobe, or possibly basing it on the docker storage driver.

markgoddard reviewed (changes requested):
Looks like a good start at this.

jovial committed:
Upstream compatability

* The privilege exec flag was removed.
* port_type was renamed to interface_type

jovial committed:
include interface_type in dictionary key

jovial committed:
Add oc_oob_interface to seed

TASK [Validate network interface configuration] *************************************************************************************************************
failed: [sib-seed] (item=oc_oob) => {"failed": true, "item": "oc_oob", "msg": "Network interface validation failed - no interface configured for oc_oob. This should be configured via 'oc_oob_interface'.\n"}

jovial committed:
force network allocation from design document for seed

jovial committed:
Fixups for network allocation pools

* subnets end at 254
* some addresses are reserved for switch and default gateway

jovial committed:
Add system_name_llc to inspector processing hooks

If you add system_name_llc here it will add the switch hostname to ironic ports' local_link_connection/switch_info, then you can refer to switches by hostname in NGS ratherthan MAC.

jovial committed:
move nova-compute-ironic.conf to proper location

jovial committed:
Add kolla-ansible autogenerated passwords

Added first run of the kolla-ansible.yml playbook

jovial committed:
Remove mlnx_sdn_assist from mechanism drivers

Not currently using infiniband.

jovial committed:
Seperate physical networks

* sib_ctl_physical_network: General purpose 1G network
* sib_mgmt_physical_network: 1G Management network

jovial committed:
Explicitly set docker registry

Using the ip from network-allocations.yml does not seem to work

jovial committed:
Add control plane provisioning network gateway

jovial committed:
use static network configuration for overcloud provisioning interface on seed

jovial committed:
Add public network

This defines the network used for the public openstack APIS.
The virtual IP is used for the user APIs. This is also the IP
used for Horizon (OpenStack dashboard)

jovial committed:
Workaround git protocol being blocked by external firewall

https is unrestricted in this deployment environment

jovial committed:
Update network allocations

This is after: kayobe overcloud inventory discover

jovial committed:
Fix copy and paste error

The network is called public not public_provisioning

jovial committed:
Manually sync network-allocation

This is to match the deployment specification

jovial committed:
Add Ironic node group mappings

These are used for kayobe overcloud inventory discover
for the auto generated etc/kayobe/inventory/overcloud
file

jovial committed:
Remove controller from storage-mgmt network

jovial committed:
Add autogenerated inventory file

This was generated from: kayobe overcloud inventory discover

jovial committed:
Add storage group to to overcloud

This is so the ceph nodes can be provisioned by
kayobe overcloud provision

oneswig committed:
Add configuration for trunk links for the 6224

A trunk to the PowerConnect 2884 BMC switch and
a trunk to the external ProCurve swgenebio switch.

oneswig committed:
Hypervisors should not require membership of the public network

oneswig committed:
first-draft config data for storage

jovial committed:
Fix indentation

jovial committed:
Explicitly declare no external networks

otherwise you get issues when doing:

kayobe overcloud host configure

with a missing external interface declaration

jovial committed:
Add compute network inteface definitions

jovial committed:
Change internal and public vips

The previous addresses are reserved for use with internal
routers

jovial committed:
Add stig and stackhpc@seed public keys to list of authorized users

jovial committed:
Fix kolla_neutron_ml2_mechanism_drivers

jovial committed:
fix up stackhpc pub key

jovial committed:
Fix kolla build blocks

oneswig reviewed (approved):
So far so good.

markgoddard commented on etc/kayobe/inventory/group_vars/ctl-switches:
```diff
@@ -0,0 +1,173 @@
+switch_interface_config_trunk_bmcs:
+  - "switchport mode trunk"
+  - "switchport trunk allowed vlan add {{ oc_oob_vlan }}"
+  - "switchport trunk allowed vlan add {{ wl_oob_vlan }}"
```
Do these default to tagged? I'm not sure it would work having multiple untagged VLANs on a trunk.

markgoddard commented on etc/kayobe/inventory/group_vars/ctl-switches:
```diff
@@ -0,0 +1,173 @@
+switch_interface_config_trunk_ext:
+  - "switchport mode trunk"
+  - "switchport trunk allowed vlan add {{ oc_oob_vlan }}"
+  - "switchport trunk allowed vlan add {{ oc_provision_vlan }}"
```
As above.

markgoddard commented on etc/kayobe/ipa.yml:
```diff
@@ -46,6 +53,11 @@
+ipa_build_dib_git_elements:
+  - repo: "https://github.com/stackhpc/stackhpc-image-elements"
+    local: "{{ source_checkout_path }}/stackhpc-image-elements"
+    version: "use-https"
```
Need to merge this and use master? Or a tag?
